samples/kernel/metairq_dispatch/src/main.c

```diff
@@ -191,7 +197,7 @@ static void thread_fn(void *p1, void *p2, void *p3)
 		 * and get queued.
 		 */
 		while (k_cycle_get_32() - start < m.proc_cyc) {
-			for (volatile int i = 0; i < 100; i++) {
+			for (volatile int i = 0; i < SPIN_TIMES; i++) {
```

Comment:
This doesn't increase the time spent waiting, it just decreases the frequency with which the timer is polled.  We wait for `m.proc_cyc` in either case.

If this is fixing something, the root cause has to be elsewhere.

Reply:
You are absolutely right. This root cause is due to race on record_latencies subroutine, add timeout here is not a right way to go. Correct it. thanks for pointing this out.